tokio-io/src/io/chain.rs

```diff
@@ -133,10 +125,16 @@ where
     }
 
     fn consume(self: Pin<&mut Self>, amt: usize) {
-        if !self.done_first {
-            self.first().consume(amt)
+        let me = self.project();
+        if !*me.done_first {
+            me.first.consume(amt)
         } else {
-            self.second().consume(amt)
+            me.second.consume(amt)
         }
     }
 }
+
+#[test]
+fn assert_unpin() {
```

Comment:
Oh, I forgot to surround this with a `#[cfg(test)] mod tests {}` module. I will push a commit for this. (but, in fact, it seems to work properly without that module...)